tensorboard/plugins/profile/pod_viewer/stack_bar_chart/stack-bar-chart.ts

```diff
@@ -0,0 +1,203 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+namespace pod_viewer_stack_bar_chart {
+
+Polymer({
+  is: 'stack-bar-chart',
+  properties: {
+    id: {
+      type: String,
+    },
+    data: {
+      type: Object,
+      notify: true,
+      observer: 'dataChanged_',
+    },
+    ready_: {
+      type: Boolean,
+      value: false,
+    },
+    stackEle: {
+      type: Array,
+      notify: true,
+      observer: 'onStackEleChanged_',
+    },
+    xDomainFunc: {
+      type: Object,
+      notify: true,
+    },
+    active: {
+      type: Object,
+      notify: true,
+    },
+  },
+  /**
+   * Main function to draw a stacked bar chart.
+   */
+  stackBarChart: function(data) {
+    if (!data || !this.ready_ || this.stackEle.length == 0) {
+      return;
+    }
+    d3.selectAll('#' + this.id + ' g > *').remove();
+    d3.select('#' + this.id + ' svg').remove();
+    d3.select('#' + this.id + '.svg-container').remove();
+    const stackKey = this.stackEle.map((d) => d.key);
+    const stackLabel = this.stackEle.map((d) => d.label);
+    const margin = {top: 20, right: 20, bottom: 30, left: 100};
+    const width = 1600 - margin.left - margin.right;
+    const height = 300 - margin.top - margin.bottom;
+    const barWidth = 50;
+    const xScaleRange = data.length * barWidth;
+    let xScale = d3.scaleBand().range([0, xScaleRange]).padding(0.4);
+    let yScale = d3.scaleLinear().range([height, 0]);
+    let colorScale = d3.scaleOrdinal<number, string>(d3.schemeCategory10)
+                       .domain([0, 19]);
+    let svg = d3.select('#' + this.id + ' #chart')
+                  .append('svg')
+                  .attr(
+                      'width',
+                      Math.max(width, xScaleRange + margin.left + margin.right))
+                  .attr('height', height + margin.top + margin.bottom)
+                  .append('g')
+                  .attr(
+                      'transform',
+                      'translate(' + margin.left + ',' + margin.top + ')');
+    let stack = d3.stack()
+                    .keys(stackKey)
+                    .order(d3.stackOrderNone)
+                    .offset(d3.stackOffsetNone);
+    const layers = stack(data);
+    xScale.domain(data.map(this.xDomainFunc));
+    yScale.domain([0, d3.max(layers[layers.length - 1], (d) => d[0] + d[1])])
+        .nice();
+    this.drawLayers(svg, layers, xScale, yScale, colorScale);
+    this.drawAxes(svg, xScale, yScale, height);
+    this.drawLegend(svg, stackLabel, colorScale);
+  },
+  /**
+   * Draw the layers for all the bars.
+   */
+  drawLayers: function(svg, layers, xScale, yScale, colorScale) {
+    let parent = this;
+    let layer = svg.selectAll('.layer')
+                    .data(layers)
```

Comment:
Can you please only indent it +4 from the previous line? This indentation is less legible.

Reply:
Done.